### PR TITLE
fix: use dedicated sha extractions per kind of event

### PR DESCRIPTION
**Pull request by nulltoken**
Fix an issue discovered by @philsturgeon in https://github.com/protect-earth/protect.earth/pull/61/checks?check_run_id=634958740

image

### CODE REVIEW

File: src/octokit.ts [hunk 1]
```diff
@@ -36,20 +36,39 @@ export interface IRepositoryInfo {
   sha: string;
 }
 
+const extractSha = (eventName: string, event: any): E.Either<Error, string> => {
+  return E.tryCatch(() => {
+    switch (eventName) {
+      case 'pull_request':
+        return event.pull_request.head.sha;
```
Comment (XVincentX) [inline src/octokit.ts:43]:
```suggestion
        return E.right(event.pull_request.head.sha);
```

File: src/octokit.ts [hunk 2]
```diff
@@ -36,20 +36,39 @@ export interface IRepositoryInfo {
...
+      case 'push':
+        return event.after;
```
Comment (XVincentX) [inline src/octokit.ts:45]:
```suggestion
        return E.right(event.after);
```

File: src/octokit.ts [hunk 3]
```diff
@@ -36,20 +36,39 @@ export interface IRepositoryInfo {
...
-    TE.map(event => {
-      const { repository, after } = event;
-      const {
-        owner: { login: owner },
-      } = repository;
-      const { name: repo } = repository;
-      return { owner, repo, eventName, sha: after };
-    })
+    TE.chain(event =>
+      pipe(
```
Comment (XVincentX) [inline src/octokit.ts:59]: You do not need to pipe, Chain and map just after

File: src/octokit.ts [hunk 4]
```diff
@@ -36,20 +36,39 @@ export interface IRepositoryInfo {
...
 
+const extractSha = (eventName: string, event: any): E.Either<Error, string> => {
+  return E.tryCatch(() => {
+    switch (eventName) {
+      case 'pull_request':
+        return event.pull_request.head.sha;
+      case 'push':
+        return event.after;
+      default:
+        throw new Error(`Unsupported event '${eventName}'`);
```
[the remark this answers is not in the document]
Comment (nulltoken) [reply src/octokit.ts]: @XVincentX Fixed

File: src/octokit.ts [hunk 5]
```diff
@@ -36,20 +36,39 @@ export interface IRepositoryInfo {
   sha: string;
 }
 
+const extractSha = (eventName: string, event: any): E.Either<Error, string> => {
+  return E.tryCatch(() => {
```
[the remark this answers is not in the document]
Comment (nulltoken) [reply src/octokit.ts]: @XVincentX Fixed

File: src/octokit.ts (hunk 1 above)
Comment (nulltoken) [reply src/octokit.ts]: @XVincentX Fixed

File: src/octokit.ts (hunk 2 above)
Comment (nulltoken) [reply src/octokit.ts]: @XVincentX Fixed

File: src/octokit.ts (hunk 3 above)
Comment (nulltoken) [reply src/octokit.ts]: @XVincentX I can't seem to get rid of pipe as I need `event` for both the `extractSha` and `map` calls. I might not see an obvious fp-ts construct...

**Pull request merged**